tools/Makefile

```
@@ -109,16 +91,20 @@ CSLPROF_IMPORTS=config.cmo build_path_prefix_map.cmo misc.cmo identifiable.cmo \
   pprintast.cmo \
   lexer.cmo parse.cmo
 
-$(call byte_and_opt,ocamlprof,$(CSLPROF_IMPORTS) profiling.cmo $(CSLPROF),)
+ocamlprof$(EXE): $(CSLPROF_IMPORTS) profiling.cmo $(CSLPROF)
+ocamlprof.opt$(EXE): \
+  $(call byte2native, $(CSLPROF_IMPORTS) profiling.cmo $(CSLPROF))
```

```suggestion
ocamlprof$(EXE): $(CSLPROF_IMPORTS) $(CSLPROF)
ocamlprof.opt$(EXE): \
  $(call byte2native, $(CSLPROF_IMPORTS) $(CSLPROF))

all: profiling.cmo
allopt: profiling.cmx
```
This has been wrong since #587 introduced it - `profiling.cmo` should always been in the third argument. It needs to built (and installed) for `ocamlcp` and `ocamloptp`, but not linked into `ocamlprof`

---

Done